ambry-api/src/main/java/com/github/ambry/config/CompressionConfig.java

```diff
@@ -0,0 +1,289 @@
+    minimalSourceDataSizeInBytes = DEFAULT_MINIMAL_DATA_SIZE_IN_BYTES;
+    minimalCompressRatio = DEFAULT_MINIMAL_COMPRESS_RATIO;
+    compressContentTypes = parseContentTypes(DEFAULT_CONTENT_TYPES_COMPRESSIBLE, DEFAULT_CONTENT_TYPES_INCOMPRESSIBLE);
+    compressContentPrefixes = parseContentPrefixes(DEFAULT_CONTENT_PREFIXES_COMPRESSIBLE,
+        DEFAULT_CONTENT_PREFIXES_INCOMPRESSIBLE);
+  }
+
+  /**
+   * Construct a compression config instance using a specific verifiable properties.
+   * @param verifiableProperties The verifiable properties.
+   */
+  public CompressionConfig(VerifiableProperties verifiableProperties) {
+    isCompressionEnabled = verifiableProperties.getBoolean(CONFIG_COMPRESSION_ENABLED, DEFAULT_COMPRESSION_ENABLED);
+    isSkipWithContentEncoding = verifiableProperties.getBoolean(CONFIG_SKIP_CONTENT_ENCODING,
+        DEFAULT_SKIP_CONTENT_ENCODING);
+    algorithmName = verifiableProperties.getString(CONFIG_ALGORITHM_NAME, DEFAULT_ALGORITHM_NAME);
+    compressOtherContentTypes = verifiableProperties.getBoolean(CONFIG_COMPRESS_OTHER_CONTENT_TYPE,
+        DEFAULT_COMPRESS_OTHER_CONTENT_TYPE);
+    minimalSourceDataSizeInBytes = verifiableProperties.getInt(CONFIG_MINIMAL_DATA_SIZE_IN_BYTES,
+        DEFAULT_MINIMAL_DATA_SIZE_IN_BYTES);
+    minimalCompressRatio = verifiableProperties.getDouble(CONFIG_MINIMAL_COMPRESS_RATIO,
+        DEFAULT_MINIMAL_COMPRESS_RATIO);
+
+    // Build the content-type map.
+    String compressibleContentTypes = verifiableProperties.getString(CONFIG_CONTENT_TYPES_COMPRESSIBLE,
+        DEFAULT_CONTENT_TYPES_COMPRESSIBLE);
+    String incompressibleContentTypes = verifiableProperties.getString(CONFIG_CONTENT_TYPES_INCOMPRESSIBLE,
+        DEFAULT_CONTENT_TYPES_INCOMPRESSIBLE);
+    compressContentTypes = parseContentTypes(compressibleContentTypes, incompressibleContentTypes);
+
+    // build the content-type prefix map.
+    String compressibleContentPrefixes = verifiableProperties.getString(CONFIG_CONTENT_PREFIXES_COMPRESSIBLE,
+        DEFAULT_CONTENT_PREFIXES_COMPRESSIBLE);
+    String incompressibleContentPrefixes = verifiableProperties.getString(CONFIG_CONTENT_PREFIXES_INCOMPRESSIBLE,
+        DEFAULT_CONTENT_PREFIXES_INCOMPRESSIBLE);
+    compressContentPrefixes = parseContentPrefixes(compressibleContentPrefixes, incompressibleContentPrefixes);
+  }
+
+  /**
+   * Combine the compressible and incompressible list of content-types into a map of content-type to boolean.
+   *
+   * @param compressibleContentTypes List of compressible content-types.
+   * @param incompressibleContentTypes List of incompressible content-types.
+   * @return A map from content-type to boolean (compressible or incompressible).
+   */
+  private Map<String, Boolean> parseContentTypes(String compressibleContentTypes, String incompressibleContentTypes) {
+    Map<String, Boolean> result = new HashMap<>();
+    addContentTypesToMap(result, compressibleContentTypes, true);
```

Comment:
some check to ensure they are mutually exclusive ?

Reply:
In this case, there is no need for exclusivity.  Instead it's import to specify the order the rules are applied.  As described in comment, the order is:
- full content-type check.
- content-type prefix check.
- other content-type setting.

For example, what if I want to block everything image/* except image/bmp?  Then add "image" to incompressible prefix, and add "image/bmp" to compressible content-type.